src/NServiceBus.Core/Sagas/SagaMetadata.cs

```diff
@@ -110,5 +119,273 @@ public bool TryGetFinder(string messageType, out SagaFinderDefinition finderDefi
         {
             return sagaFinders.TryGetValue(messageType, out finderDefinition);
         }
+
+        /// <summary>
+        /// Creates a <see cref="SagaMetadata" /> from a specific Saga type.
+        /// </summary>
+        /// <param name="sagaType">A type representing a Saga. Must be a non-generic type inheriting from <see cref="NServiceBus.Saga.Saga" />.</param>
+        /// <returns>An instance of <see cref="SagaMetadata" /> describing the Saga.</returns>
+        public static SagaMetadata Create(Type sagaType)
+        {
+            return Create(sagaType, new List<Type>(), new Conventions());
+        }
+
+        /// <summary>
+        /// Creates a <see cref="SagaMetadata" /> from a specific Saga type.
+        /// </summary>
+        /// <param name="sagaType">A type representing a Saga. Must be a non-generic type inheriting from <see cref="NServiceBus.Saga.Saga" />.</param>
+        /// <param name="availableTypes">Additional available types, used to locate saga finders and other related classes.</param>
+        /// <param name="conventions">Custom conventions to use while scanning types.</param>
+        /// <returns>An instance of <see cref="SagaMetadata" /> describing the Saga.</returns>
+        public static SagaMetadata Create(Type sagaType, IEnumerable<Type> availableTypes, Conventions conventions)
```

Comment:
Not sure if this fits within the scope of this PR so feel free to ignore. I am concern that the process of constructing the metadata object is tied to this object. It's like a car engine knowing how to construct itself.